sql/commitment-funding.sql

```diff
@@ -0,0 +1,667 @@
+		plannedcommit_total numeric
+	);
+	
+	CREATE TABLE project_source (
+		ccp_version text,
+		m_a_proj_id text,
+		m_agency_acro text,
+		m_agency char(3),
+		m_agency_name text,
+		description text,
+		proj_id text,
+		min_date date,
+		max_date date,
+		type_category text,
+		plannedcommit_ccnonexempt numeric,
+		plannedcommit_ccexempt numeric,
+		plannedcommit_citycost numeric,
+		plannedcommit_nccstate numeric,
+		plannedcommit_nccfederal numeric,
+		plannedcommit_nccother numeric,
+		plannedcommit_noncitycost numeric,
+		plannedcommit_total numeric,
+		adopt_ccnonexempt numeric,
+		adopt_ccexempt numeric,
+		adopt_citycost numeric,
+		adopt_nccstate numeric,
+		adopt_nccfederal numeric,
+		adopt_nccother numeric,
+		adopt_noncitycost numeric,
+		adopt_total numeric,
+		allocate_ccnonexempt numeric,
+		allocate_ccexempt numeric,
+		allocate_citycost numeric,
+		allocate_nccstate numeric,
+		allocate_nccfederal numeric,
+		allocate_nccother numeric,
+		allocate_noncitycost numeric,
+		allocate_total numeric,
+		commit_ccnonexempt numeric,
+		commit_ccexempt numeric,
+		commit_citycost numeric,
+		commit_nccstate numeric,
+		commit_nccfederal numeric,
+		commit_nccother numeric,
+		commit_noncitycost numeric,
+		commit_total numeric,
+		spent_ccnonexempt numeric,
+		spent_ccexempt numeric,
+		spent_citycost numeric,
+		spent_nccstate numeric,
+		spent_nccfederal numeric,
+		spent_nccother numeric,
+		spent_noncitycost numeric,
+		spent_total numeric,
+		spent_total_checkbooknyc numeric
+	);
+	
+COMMIT;
+
+-- Populate source tables
+BEGIN;
+	COPY project_source FROM 'cpdb_projects_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+BEGIN;
+	COPY commitment_source FROM 'cpdb_planned_commitments_23adopt.csv' DELIMITER ',' CSV HEADER;
+COMMIT;
+
+-- Structure of target tables
+-- Right now, we are using data as primary keys. We should evaluate when they should be replaced with separate ids
+BEGIN;
+	CREATE TABLE managing_code (
+		id char(3) PRIMARY KEY
+	);
+
+	-- Should likely create an id distinct from initials
+	CREATE TABLE agency (
+		initials text PRIMARY KEY,
+		name text
+	);
+	
+	CREATE TYPE project_category AS ENUM (
+		'Fixed Asset',
+		'Lump Sum',
+		'ITT, Vehicles, and Equipment'
+	);
+		
+	CREATE TABLE project (
+		PRIMARY KEY (managing_code, id),
+    	managing_code char(3) references managing_code(id),
+		id text,
+		managing_agency text references agency(initials),
+		description text,
+		min_date date,
+		max_date date,
+		category project_category		
+	);
+
+	CREATE TABLE project_checkbook(
+		id uuid PRIMARY KEY,
+		managing_code char(3),
+		project_id text,
+		FOREIGN KEY (managing_code, project_id) references project(managing_code, id),
+		value numeric
+	);
+
+	CREATE TYPE fund_category AS ENUM (
+		'city-non-exempt',
+		'city-exempt',
+		'city-cost',
+		'non-city-state',
+		'non-city-federal',
+		'non-city-other',
+		'non-city-cost',
+		'total'
+	);
+
+	CREATE TYPE project_fund_stage AS ENUM (
```

Comment:
I think using an enum here is great but it does make me wonder if these stages are "ordinal". If they are, and we ever have use cases that need that relationship, this could be a good candidate to convert to a look up table. In any case, not something we should worry about now.